Symfony/CS/Fixer/Contrib/ClassElementsOrderFixer.php

```diff
@@ -0,0 +1,305 @@
+        'method_public',
+        'method_protected',
+        'method_private',
+    );
+
+    private $config;
+
+    public function __construct()
+    {
+        $this->configure(null);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        $configuration = $configuration ?: self::$defaultOrder;
+
+        $this->config = array();
+
+        foreach ($configuration as $pos => $type) {
+            $this->config[$type] = $pos;
+        }
+
+        $lastPosition = count($configuration);
+        foreach (self::$types as $type => $parents) {
+            if (isset($this->config[$type])) {
+                continue;
+            }
+
+            if ($parents && null !== $pos = $this->findConfigurationPosition($parents)) {
+                $this->config[$type] = $pos;
+                continue;
+            }
+
+            if (!in_array($type, array('construct', 'phpunit'), true)) {
+                $this->config[$type] = $lastPosition;
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound(Token::getClassyTokenKinds());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        for ($i = 1, $count = $tokens->count(); $i < $count; ++$i) {
+            if (!$tokens[$i]->isClassy()) {
+                continue;
+            }
+
+            $i = $tokens->getNextTokenOfKind($i, array('{'));
+            $elements = $this->getElements($tokens, $i);
+
+            if (!$elements) {
+                continue;
+            }
+
+            $sorted = $this->sortElements($elements);
+            $endIndex = $elements[count($elements) - 1]['end'];
+
+            if ($sorted !== $elements) {
+                $this->sortTokens($tokens, $i, $endIndex, $sorted);
+            }
+
+            $i = $endIndex;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Orders the elements of classes/interfaces/traits.';
+    }
+
+    private function findConfigurationPosition(array $types)
+    {
+        foreach ($types as $type) {
+            if (isset($this->config[$type])) {
+                return $this->config[$type];
+            }
+        }
+
+        foreach ($types as $type) {
+            $parents = self::$types[$type];
+
+            if ($parents && $pos = $this->findConfigurationPosition($parents)) {
+                return $pos;
+            }
+        }
+    }
+
+    private function getElements(Tokens $tokens, $startIndex)
+    {
+        ++$startIndex;
+        $elements = array();
+
+        while (true) {
+            $element = array(
+                'start' => $startIndex,
+                'visibility' => 'public',
+                'static' => false,
+            );
+            for ($i = $startIndex; ; ++$i) {
```

Comment:
Before I started to implement this part I looked at the method. IMO it would need more code to use that method than without using it, caused by special needs of this fixer. And the tokens would be traversed more often.

(at the moment this fixer traverse each token only once..)